### PR TITLE
[Do not merge] use line protocol to consume from kafka

### PR DESCRIPTION
**Pull request by replay**

### CODE REVIEW

**Comment by replay** [comment]
Needed to unvendor `gopkg.in/raintank/schema.v0` because otherwise the types `[]*"github.com/raintank/metrictank/vendor/gopkg.in/raintank/schema.v1".MetricData` and `[]*"gopkg.in/raintank/schema.v1".MetricData` do not match

`./metrics.go:114: cannot use buf (type []*"github.com/raintank/metrictank/vendor/gopkg.in/raintank/schema.v1".MetricData) as type []*"gopkg.in/raintank/schema.v1".MetricData in argument to msg.CreateMsg`

File: cmd/mt-replicator-via-tsdb/metrics.go [hunk 1]
```diff
@@ -142,14 +144,30 @@ func (r *MetricsReplicator) Consume() {
 				}
 				return
 			}
-			md := &schema.MetricData{}
-			_, err := md.UnmarshalMsg(m.Value)
-			if err != nil {
-				log.Error(3, "kafka-mdm decode error, skipping message. %s", err)
-				continue
+			for _, line := range bytes.Split(m.Value, []byte("\n")) {
+				key, val, ts, err := m20.ValidatePacket(
```
Comment (woodsaj) [inline cmd/mt-replicator-via-tsdb/metrics.go]: you should probably make sure that `line != ""`

Comment (replay) [inline cmd/mt-replicator-via-tsdb/metrics.go]: right :+1:

**Comment by Dieterbe** [comment]
> Needed to unvendor gopkg.in/raintank/schema.v0 because otherwise the types []*"github.com/raintank/metrictank/vendor/gopkg.in/raintank/schema.v1".MetricData and []*"gopkg.in/raintank/schema.v1".MetricData do not match

this seems bad. all dependencies should be vendored.  i'm not sure what the exact problem/solution are though. but i ran into a similar problem a while back : https://gophers.slack.com/archives/C0M5YP9LN/p1473833774000084 reading that discussion should be helpful here.

**Comment by replay** [comment]
@Dieterbe sorry, i forgot to mention in the PR description that this has never been supposed to get merged. that's why i didn't hesitate to unvedor pkgs

**Comment by Dieterbe** [comment]
do we need this for anything? can we delete?

**Comment by replay** [comment]
@woodsaj do we still need that? I thought there was one specific use case why we needed that, but I forgot if it still exists

**Comment by woodsaj** [comment]
This was for a one-time use to help a customer consume raw metrics from their internal kafka and write them to hosted-metrics.  I think they were using carbon-relay, but were seeing a high number of out-of-order metrics, so we suggested this tool to help them out.  But the re-order buffer would also have solved their problem.  

So we dont need this.

**Pull request closed without merging**